### DISCUSSION TITLE
Groovy-language-server, duplicate definition warning in eglot

### DISCUSSION THREAD

**Discussion by triHims** [joaotavora/eglot#1260] [Q&A] [answered]
```
(use-package eglot

:ensure t

:defer t

:config

(setq read-process-output-max (* 1024 1024))

(push :documentHighlightProvider eglot-ignored-server-capabilities)

;;Adding groovy server

(add-to-list 'eglot-server-programs \(groovy-mode . ( ,local-java11-path "-jar" ,(expand-file-name (concat local-abs-config-path local-abs-custom-utils "groovy-language-server/groovy-language-server-all.jar")))))`

)
```

This is my eglot config for groovy-language-server.
I load my classpath by .dir-locals.el 
```
((groovy-mode
  . (
     (eglot-workspace-configuration
      . (:groovy (:classpath
[
 "/Volumes/WORK/Stuff/persuade/johndeere-api/libs/SparkJDBC41.jar"
....]
		  )))))
 )
```

Please help me with this.

**Comment by manuel-uberti** [comment]
In order to understand the actual problem, could you try following the instructions at https://joaotavora.github.io/eglot/#Troubleshooting-Eglot?

**Comment by triHims** [comment]
@manuel-uberti 
Thank you for the response. 
After the fresh start i start seeing the duplicate-warning. (No duplicate file exists, so the warning is irrelevant)
Attached a event log. There is no error log the buffer was blank.
Please find the attached logs.

[Events-buffer.log](https://github.com/joaotavora/eglot/files/12237512/Events-buffer.log)

**Reply by triHims** [reply]
this is the message which i get in the diagnostic. 
Invalid duplicate class definition of class com.project.work.lib.service.impl.WorkUploadServiceImpl : The sources /Volumes/WORK/Stuff/Project/work-api/src/main/groovy/com/project/work/lib/service/impl/WorkUploadServiceImpl.groovy and /Volumes/WORK/Stuff/Project/work-api/src/main/groovy/com/project/work/lib/service/impl/WorkUploadServiceImpl.groovy each contain a class with the name com.project.work.lib.service.impl.WorkUploadServiceImpl.\n @ line 11, column 1.")]))

**Reply by manuel-uberti** [reply]
Did you follow the link I posted above? Did you try from `emacs -Q`? Can you try to provide a minimal recipe to reproduce the problem?

Note that I am not that maintainer of this project, I am just trying to have you giving them _all_ the information needed in order to understand the problem.

**Reply by joaotavora** [reply]
> Note that I am not that maintainer of this project, I am just trying to have you giving them all the information needed in order to understand the problem.

Yes, and it is much appreciated.  Thanks @manuel-uberti

**Reply by triHims** [reply]
Thank you for the response.
 I will try to create a project that can reproduce the errors and try to provide all the details surrounding it.

**Reply by triHims** [reply, marked as answer]
Unfortunately this error is happening in a particular project and does not happen in others.
Due to this i will am unable to provide a dummy project to replicate it properly. 
Hence closing this. 
Thanks for your time.

**Reply by joaotavora** [reply]
Ok, it also means that you can "bisect" your project. I.e. remove parts of as to incrementally make it look more like the dummy project where the problem _doesn't_ happen.  At a certain point you'll notice you remove a certain part of the problematic project that makes the error go away, so that's probably the part you should focus on.  Then you can post your results here, if you want.

**Reply by triHims** [reply]
Great. I'd try that. Thank you.
